scripts/generate-sitemap.js

```diff
@@ -14,28 +17,90 @@ const siteMetadata = require('../data/siteMetadata')
     '!pages/api',
   ])
 
+  const { locales, defaultLocale } = i18nConfig
+
+  const pagesWithLoc = pages
+    .map((page) => {
+      if (page.includes('pages')) {
+        return locales.map((locale) => [page, locale])
+      }
+
+      if (page.includes('data') || page.includes('.xml')) {
+        for (let i = 0; i < locales.length; i++) {
+          if (page.includes(`.${locales[i]}.`)) {
+            return [[page, locales[i]]]
+          }
+        }
+        return [[page, defaultLocale]]
+      }
+
+      throw new Error('Sitemap case missing, please check scripts/generate-sitemap.js')
+    })
+    .flat()
+    .map(([page, loc]) => [
+      (page =
+        (loc !== defaultLocale ? `/${loc}` : '') +
+        page
+          .replace('pages/', '/')
+          .replace('data/blog', '/blog')
+          .replace('public/', '/')
+          .replace('.js', '')
+          .replace('.mdx', '')
+          .replace('.md', '')
+          .replace(`.${loc}`, '')
+          .replace('/feed', '')
+          .replace('.xml', '')),
+      loc,
+      false, // Indicate if the element is already present or not
+    ])
+
   const sitemap = `
-        <?xml version="1.0" encoding="UTF-8"?>
-        <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
-            ${pages
-              .map((page) => {
-                const path = page
-                  .replace('pages/', '/')
-                  .replace('data/blog', '/blog')
-                  .replace('public/', '/')
-                  .replace('.js', '')
-                  .replace('.mdx', '')
-                  .replace('.md', '')
-                  .replace('/feed.xml', '')
-                const route = path === '/index' ? '' : path
-                if (page === `pages/404.js` || page === `pages/blog/[...slug].js`) {
+  <?xml version="1.0" encoding="UTF-8"?>
+  <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:xhtml="http://www.w3.org/1999/xhtml"
+      xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9
+      http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd
+      http://www.w3.org/1999/xhtml
+      http://www.w3.org/2002/08/xhtml/xhtml1-strict.xsd">
+            ${pagesWithLoc
+              .map(([path, loc, alreadyPresent]) => {
+                // @todo: Can you check especially here ?
+                const route = path.includes('/index') ? path.replace('/index', '') : path
+                if (
+                  path.includes(`/404.js`) ||
```

Comment:
`path.includes(`/404`) || alreadyPresent` would do. The path would not contain `.js` so currently the 404 page is still in the sitemap